src/api/utils.jl

```diff
@@ -56,3 +56,28 @@ function _bulkify(payload::Vector{String})
 
   join(payload, "\n")
 end
+
+function extract_options(arguments::AbstractDict)::AbstractDict
+  options = Dict{Symbol, Any}()
+
+  extract_option_if_exist(option_name::Symbol) = begin
```

Comment:
Not sure that this is a good idea due to performance. It would be good to implement a common function `extract_option_if_exist!` with the argument `options`.

Reply:
fixed